### PR TITLE
Log redirection

### PR DESCRIPTION
**Pull request by torusrxxx**
Redirect log to a text file or a pipe object.

---
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/x64dbg/849)

### CODE REVIEW

**Comment by mrexodia** [comment]
I would like to avoid windows api in the gui if at all possible. Can this be done with qfile or fopen?

**Comment by torusrxxx** [comment]
I have looked through the Qt documentation. QLocalSocket provides named pipe support. This implies QFile is not intended for accessing named pipes.

**Comment by mrexodia** [comment]
What about `fopen`?

**Comment by torusrxxx** [comment]
`fopen` seems to support named pipe. I have never tested it though, only writing to a text file is tested.

File: src/gui/Src/Gui/LogView.cpp [hunk 1]
```diff
@@ -87,6 +112,29 @@ void LogView::clearLogSlot()
...
+{
+    if(logRedirection != NULL)
+        fclose(logRedirection);
+    logRedirection = NULL;
+    BrowseDialog browse(this, tr("Redirect log to file"), tr("Enter the file to which you want to redirect log messages."), tr("Log files(*.txt);;All files(*.*)"), QCoreApplication::applicationDirPath(), true);
+    if(browse.exec() == QDialog::Accepted)
+    {
+        logRedirection = _wfopen(browse.path.toStdWString().c_str(), L"ab");
+        if(logRedirection == NULL)
+            addMsgToLogSlot(tr("CreateFile() failed. Log will not be redirected to %1.\n").arg(browse.path));
```
Comment (justanotheranonymoususer) [inline src/gui/Src/Gui/LogView.cpp]: `CreateFile` -> `fopen`.

**Comment by mrexodia** [comment]
Great, thanks!

**Pull request merged**